source/common/http/utility.h

```diff
@@ -664,6 +680,13 @@ convertCoreToRouteRetryPolicy(const envoy::config::core::v3::RetryPolicy& retry_
  */
 bool isSafeRequest(const Http::RequestHeaderMap& request_headers);
 
+/**
+ * @param value: the value of the referer header field
+ * @return true if the given value conforms to RFC specifications
+ * https://www.rfc-editor.org/rfc/rfc7231#section-5.5.2
+ */
+bool isValidRefererValue(absl::string_view value);
```

Comment:
Please add unit tests for this method in the test/common/http/utility_test.cc suite.